cql3/restrictions/statement_restrictions.cc

```diff
@@ -1085,23 +1011,35 @@ value_set intersection(value_set a, value_set b, const abstract_type* type) {
 
 bool is_satisfied_by(const binary_operator& opr, const column_value_eval_bag& bag) {
     return std::visit(overloaded_functor{
-            [&] (const std::vector<column_value>& cvs) {
+            [&] (const column_value& col) {
                 if (*opr.op == operator_type::EQ) {
-                    return equal(opr.rhs, cvs, bag);
+                    return equal(*opr.rhs, col, bag);
                 } else if (*opr.op == operator_type::NEQ) {
-                    return !equal(opr.rhs, cvs, bag);
+                    return !equal(*opr.rhs, col, bag);
                 } else if (opr.op->is_slice()) {
-                    return limits(opr, bag);
+                    return limits(col, *opr.op, *opr.rhs, bag);
                 } else if (*opr.op == operator_type::CONTAINS) {
-                    return contains(opr.rhs->bind_and_get(bag.options), cvs, bag);
+                    return contains(col, opr.rhs->bind_and_get(bag.options), bag);
                 } else if (*opr.op == operator_type::CONTAINS_KEY) {
-                    return contains_key(cvs, opr.rhs->bind_and_get(bag.options), bag);
+                    return contains_key(col, opr.rhs->bind_and_get(bag.options), bag);
                 } else if (*opr.op == operator_type::LIKE) {
-                    return like(cvs, *opr.rhs, bag);
+                    return like(col, to_bytes_opt(opr.rhs->bind_and_get(bag.options)), bag);
+                } else if (*opr.op == operator_type::IN) {
+                    return is_one_of(col, *opr.rhs, bag);
+                } else {
+                    throw exceptions::unsupported_operation_exception(format("Unhandled binary_operator: {}", opr));
+                }
+            },
+            [&] (const std::vector<column_value>& cvs) {
+                if (*opr.op == operator_type::EQ) {
+                    return equal(*opr.rhs, cvs, bag);
+                } else if (opr.op->is_slice()) {
+                    return limits(cvs, *opr.op, *opr.rhs, bag);
                 } else if (*opr.op == operator_type::IN) {
                     return is_one_of(cvs, *opr.rhs, bag);
                 } else {
-                    throw exceptions::unsupported_operation_exception("Unhandled binary_operator");
```

Comment:
I see multi-column like is gone.

Reply:
Yeah, multi-column LIKE was a fancy generalization that the upstream might allow one day.  We don't need it right now.

Reply:
Is it in any branch? (4.2?)

Reply:
> Is it in any branch? (4.2?)

No, it's a part of the big rewrite b0698dfb38afacbd3daf8f64ec83e63ba0ce0cd9.